src/components/Toggle/tests/Toggle.test.tsx

```diff
@@ -0,0 +1,173 @@
+import * as React from 'react';
+import {noop} from '@shopify/javascript-utilities/other';
+import {mountWithAppProvider, findByTestID} from 'test-utilities';
+import Labelled from '../../Labelled';
+import TextContainer from '../../TextContainer';
+import Icon from '../../Icon';
+import Toggle from '../Toggle';
+
+describe('<Toggle />', () => {
+  const mockProps = {
+    label: 'label',
+    onChange: noop,
+  };
+
+  describe('role', () => {
+    it('gets set to checkbox', () => {
+      const toggle = mountWithAppProvider(<Toggle {...mockProps} />);
+      const input = findByTestID(toggle, 'ToggleInput');
+
+      expect(input.prop('role')).toBe('checkbox');
+    });
+  });
+
+  describe('checked', () => {
+    it('sets aria-checked if true', () => {
+      const toggle = mountWithAppProvider(<Toggle {...mockProps} checked />);
+      const input = findByTestID(toggle, 'ToggleInput');
+
+      expect(input.prop('aria-checked')).toBeTruthy();
+    });
+
+    it('does not set aria-checked if false', () => {
+      const toggle = mountWithAppProvider(<Toggle {...mockProps} />);
+      const input = findByTestID(toggle, 'ToggleInput');
+
+      expect(input.prop('aria-checked')).toBeFalsy();
+    });
+
+    it('is set to false by default', () => {
+      const toggle = mountWithAppProvider(<Toggle {...mockProps} />);
+      const input = findByTestID(toggle, 'ToggleInput');
+
+      expect(input.prop('checked')).toBeFalsy();
```

Comment:
```suggestion
      expect(input.prop('checked')).toBe(false);
```

Which will actually fail because the value defaults to `undefined`.